test-suite/src/main/scala/testsuite/core/MatchTest.scala

```diff
@@ -0,0 +1,51 @@
+package testsuite.core
+
+import testsuite.Assert.ok
+
+object MatchTest {
+  def main(): Unit = {
+    testFib()
+    testString()
+    testNull()
+  }
+
+  private def testFib(): Unit = {
+    def test(x: Int, expected: Int) = ok(fib(x) == expected)
+    def fib(x: Int): Int = x match {
+      case 0     => 0
+      case 1 | 2 => 1
+      case _     => fib(x - 1) + fib(x - 2)
+    }
+    test(0, 0)
+    test(1, 1)
+    test(2, 1)
+    test(3, 2)
+    test(4, 3)
+    test(5, 5)
+    test(6, 8)
+  }
+
+  private def testString(): Unit = {
+    def test(x: String, expected: String) = ok(animalSound(x) == expected)
+    def animalSound(animal: String) = animal match {
+      case "dog" => "bow-wow"
+      case "cat" => "meow"
+      case _     => "unknown"
+    }
+    test("dog", "bow-wow")
+    test("cat", "meow")
+    test("???", "unknown")
+  }
+
+  private def testNull(): Unit = {
+    def strTest(x: String) = x match {
+      case null => "null"
+      case x    => x
+    }
```

Comment:
When there aren't enough cases, the Scala compiler emits `If`s instead of `Match`es. In particular this one emits an if/then/else. Adding more cases, as follows, turns it into a real `Match`:
```suggestion
    def strTest(x: String) = x match {
      case null  => "null"
      case "foo" => "bar"
      case "bar" => "babar"
      case x     => x
    }
```

Reply:
Right, thanks, now it generates Match :+1: